ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java

```diff
@@ -712,16 +783,22 @@ public void run() {
                 progressHandle.switchToDeterminate(files.size());
 
                 updateProgress(0.0);
+                
+               taskCompletionStatus = true;
 
                 //do in transaction
                 DrawableDB.DrawableTransaction tr = taskDB.beginTransaction();
                 int workDone = 0;
                 for (final AbstractFile f : files) {
-                    if (isCancelled() || Thread.interrupted()) {
+                    if (isCancelled()) {
                         LOGGER.log(Level.WARNING, "Task cancelled: not all contents may be transfered to drawable database."); //NON-NLS
                         progressHandle.finish();
                         break;
                     }
+                    
+                    if (Thread.interrupted()) {
+                        LOGGER.log(Level.WARNING, "BulkTransferTask interrupted. Ignoring it to update the contents of drawable database."); //NON-NLS
+                    }
```

Comment:
Please write a JIRA about this - the interrupt should be equivalent to a cancellation, so we need to find out how this thread is being unexpectedly interrupted. When the JIRA is created please add a comment of the form "TODO (EUR-XXXX)" to explain why this code is the way is it.